tests/test_parse.py

```diff
@@ -29,6 +38,94 @@ def test_parse(self):
         self.assertEqual(str(msg), sample_hl7)
         self.assertEqual(str(msg), sample_hl7)
 
+    def test_parse_batch(self):
+        batch = hl7.parse_batch(sample_batch)
+        self.assertEqual(len(batch), 1)
+        self.assertIsInstance(batch[0], hl7.Message)
+        self.assertIsInstance(batch.header, hl7.Segment)
+        self.assertEqual(batch.header[0][0], "BHS")
+        self.assertEqual(batch.header[4][0], "ABCHS")
+        self.assertIsInstance(batch.trailer, hl7.Segment)
+        self.assertEqual(batch.trailer[0][0], "BTS")
+        self.assertEqual(batch.trailer[1][0], "1")
+
+    def test_parse_batch1(self):
+        batch = hl7.parse_batch(sample_batch1)
+        self.assertEqual(len(batch), 2)
+        self.assertIsInstance(batch[0], hl7.Message)
+        self.assertEqual(batch[0][0][10][0], "12334456778890")
+        self.assertIsInstance(batch[1], hl7.Message)
+        self.assertEqual(batch[1][0][10][0], "12334456778891")
+        self.assertIsInstance(batch.header, hl7.Segment)
+        self.assertEqual(batch.header[0][0], "BHS")
+        self.assertEqual(batch.header[4][0], "ABCHS")
+        self.assertIsInstance(batch.trailer, hl7.Segment)
+        self.assertEqual(batch.trailer[0][0], "BTS")
+        self.assertEqual(batch.trailer[1][0], "2")
+
+    def test_parse_batch2(self):
+        batch = hl7.parse_batch(sample_batch2)
+        self.assertEqual(len(batch), 2)
+        self.assertIsInstance(batch[0], hl7.Message)
+        self.assertEqual(batch[0][0][10][0], "12334456778890")
+        self.assertIsInstance(batch[1], hl7.Message)
+        self.assertEqual(batch[1][0][10][0], "12334456778891")
+        self.assertFalse(batch.header)
+        self.assertFalse(batch.trailer)
+
+    def test_parse_file(self):
```

Comment:
I'd consider adding test coverage around the branches of parse_file (especially non-happy path), as that is complicated logic that we have a high probability of unknowingly breaking in the future if there isn't test coverage to document the intent.

Reply:
I'll add some sad-path tests for both batch and file